fluentd/lib/fluent-plugin-collected/lib/fluent/plugin/in_collected_tail_monitor.rb

```diff
@@ -0,0 +1,140 @@
+require 'fluent/plugin/input'
+require 'fluent/plugin/in_monitor_agent'
+require 'fluent/plugin/prometheus'
+
+module Fluent::Plugin
+  # Collect log_collected_bytes_total metric from in_tail plugin.
+  class CollectedTailMonitorInput < Fluent::Plugin::Input
+    Fluent::Plugin.register_input('collected_tail_monitor', self)
+    include Fluent::Plugin::PrometheusLabelParser
+
+    helpers :timer
+
+    #ideally interval must be rotatewait > 0 ? rotatewait/2 : 1.0 and if interval >= rotate_wait we should log a warning. For rotatewait = 5 sec, interval can be set to 2 sec
+    config_param :interval, :time, default: 2
+    attr_reader :registry
+
+    MONITOR_IVARS = [
+      :tails,
+    ]
+
+    def initialize
+      super
+      @registry = ::Prometheus::Client.registry
+      # As per k8 regex for container logfile symlink ref :  https://github.com/fabric8io/fluent-plugin-kubernetes_metadata_filter/blob/master/lib/fluent/plugin/filter_kubernetes_metadata.rb#L56
+      @tag_to_kubernetes_filename_regexp_compiled = Regexp.new('var.log.containers.(?<pod_name>[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*)_(?<namespace>[^_]+)_(?<container_name>.+)-(?<docker_id>[a-z0-9]{64})\.log$')
+
+    end
+
+    def multi_workers_ready?
+      true
+    end
+
+    def configure(conf)
+      super
+      hostname = Socket.gethostname
+      expander_builder = Fluent::Plugin::Prometheus.placeholder_expander(log)
+      expander = expander_builder.build({ 'hostname' => hostname, 'worker_id' => fluentd_worker_id })
+      @base_labels = parse_labels_elements(conf)
+      @base_labels.each do |key, value|
+        unless value.is_a?(String)
+          raise Fluent::ConfigError, "record accessor syntax is not available in collected_tail_monitor"
+        end
+        @base_labels[key] = expander.expand(value)
+      end
+
+      if defined?(Fluent::Plugin) && defined?(Fluent::Plugin::MonitorAgentInput)
+        # from v0.14.6
+        @monitor_agent = Fluent::Plugin::MonitorAgentInput.new
+      else
+        @monitor_agent = Fluent::MonitorAgentInput.new
+      end
+    end
+
+    def start
+      super
+
+      @metrics = {
+        total_bytes_collected: get_counter(
+          :log_collected_bytes_total,
+          'Total bytes collected from file.'),
+      }
+      timer_execute(:in_collected_tail_monitor, @interval, &method(:update_monitor_info))
+    end
+
+
+    def update_monitor_info
+      opts = {
```

Comment:
I'm not sure if @monitor_agent.plugins_info_all(opts) can modify opts - I'd be inclined to leave it as-is to be safe.